oshdb/src/main/java/org/heigit/ohsome/oshdb/OSHDBBoundable.java

```diff
@@ -0,0 +1,52 @@
+package org.heigit.ohsome.oshdb;
+
+public interface OSHDBBoundable {
+  
+  long getMinLonLong();
+  
+  long getMinLatLong();
+  
+  long getMaxLonLong();
+  
+  long getMaxLatLong();
+  
+  default OSHDBBoundingBox getBoundingBox() {
+    return new OSHDBBoundingBox(getMinLonLong(), getMinLatLong(), getMaxLonLong(), getMaxLatLong());
+  }
+  
+  default double getMinLon() {
+    return getMinLonLong() * OSHDB.GEOM_PRECISION;
+  }
+  
+  default double getMinLat() {
+    return getMinLatLong() * OSHDB.GEOM_PRECISION;
+  }
+
+  default double getMaxLon() {
+    return getMaxLonLong() * OSHDB.GEOM_PRECISION;
+  }
+
+  default double getMaxLat() {
+    return getMaxLatLong() * OSHDB.GEOM_PRECISION;
+  }
+  
+  default boolean intersects(OSHDBBoundable otherBbox) {
+    return (otherBbox != null)
+        && (getMaxLatLong() >= otherBbox.getMinLatLong()) 
+        && (getMinLatLong() <= otherBbox.getMaxLatLong()) 
+        && (getMaxLonLong() >= otherBbox.getMinLonLong()) 
+        && (getMinLonLong() <= otherBbox.getMaxLonLong());
+  }
+  
+  default boolean isInside(OSHDBBoundingBox otherBbox) {
+    return (otherBbox != null) 
+        && (getMinLatLong() >= otherBbox.getMinLatLong()) 
+        && (getMaxLatLong() <= otherBbox.getMaxLatLong())
+        && (getMinLonLong() >= otherBbox.getMinLonLong()) 
+        && (getMaxLonLong() <= otherBbox.getMaxLonLong());
+  }
+  
+  default boolean isPoint() {
+    return getMinLonLong() == getMaxLonLong() && getMinLatLong() == getMaxLatLong();
+  }
```

Comment:
missing javadoc for these methods

```suggestion
  /** Returns true if the {@code other} boundable object intersects this. */
  default boolean intersects(OSHDBBoundable other) {
    return (other != null)
        && (getMaxLatLong() >= other.getMinLatLong())
        && (getMinLatLong() <= other.getMaxLatLong())
        && (getMaxLonLong() >= other.getMinLonLong())
        && (getMinLonLong() <= other.getMaxLonLong());
  }

  /** Returns true this boundable is inside the {@code other} object. */
  default boolean isInside(OSHDBBoundingBox other) {
    return (other != null) 
        && (getMinLatLong() >= other.getMinLatLong())
        && (getMaxLatLong() <= other.getMaxLatLong())
        && (getMinLonLong() >= other.getMinLonLong())
        && (getMaxLonLong() <= other.getMaxLonLong());
  }

  /** Returns true this boundable is degenerated to a point. */
  default boolean isPoint() {
    return getMinLonLong() == getMaxLonLong() && getMinLatLong() == getMaxLatLong();
  }
```

PS: the logic of `isInside` could be reversed and be called `contains`, to be similar to the [JTS](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/Geometry.html#contains-org.locationtech.jts.geom.Geometry-) convention.

Reply:
I like to name the methods closer to JTS. So I would rename 
- isInside to  coveredBy (even this method actually not exist in JTS Envelope but to not change the logic on the call sites)
and
- add a new method covers which is then the reverse of (isInside/coveredBy)

Thank you for your comment suggestion, I will take them and extend them to be full javadoc comments.
Like
```
  /**
   * Returns true if the {@code other} {@code OSHDBBoundable} intersects this.
   * 
   * @param other  the <code>OSHDBBoundable</code> which this <code>OSHDBBoundable</code> is
   *          being checked for intersecting
   * @return <code>true</code> if the <code>OSHDBBoundable</code>s intersect
   */
  default boolean intersects(OSHDBBoundable other) {
```